test/functional/mempool_packages.py

```diff
@@ -72,6 +74,10 @@ def run_test(self):
             value = sent_value
             chain.append(txid)
 
+        # Wait until mempool transactions have passed initial broadcast (got an inv for each one)
```

Comment:
The condition for initial broadcast is not that an inv was sent, but that a getdata was received

I think you can simply replace this with a `self.sync_mempools()` and remove the newly added p2p connection

Reply:
> I think you can simply replace this with a self.sync_mempools() and remove the newly added p2p connection

I've tried this, but it times out: `AssertionError: Mempool sync timed out after 60s`

> The condition for initial broadcast is not that an inv was sent, but that a getdata was received

Oh this is my misunderstanding. I can try to have a `wait_until` that counts the `msg_getdata`s received by `nodes[0]`?

Reply:
Oh, I missed that the second node has a stricter `limitancestorcount` set.

In that case your suggestion to wait for the getdata makes most sense, I guess.

So 

* register callback in mininode to send getdata when inv arrives
* wait for invs
* sync_with_ping (to flush all buffers)

should work

Reply:
Thank you, this is super helpful! 🙏